src/AutoRest.CSharp/Mgmt/Generation/NonLongRunningOperationWriter.cs

```diff
@@ -0,0 +1,114 @@
+        /// <param name="writer"></param>
+        /// <param name="operation"></param>
+        public static void Write(CodeWriter writer, NonLongRunningOperation operation)
+        {
+            var responseVariable = "response";
+
+            var cs = operation.Type;
+            var @namespace = cs.Namespace;
+            using (writer.Namespace(@namespace))
+            {
+                writer.WriteXmlDocumentationSummary(operation.Description);
+                var baseType = operation.ResultType != null ? new CSharpType(typeof(Operation<>), operation.ResultType) : new CSharpType(typeof(Azure.Operation));
+                var valueTaskType = operation.ResultType != null ? new CSharpType(typeof(Response<>), operation.ResultType) : new CSharpType(typeof(Response));
+                var waitForCompletionType = new CSharpType(typeof(ValueTask<>), valueTaskType);
+                var helperType = operation.ResultType != null ? new CSharpType(typeof(OperationOrResponseInternals<>), operation.ResultType) : new CSharpType(typeof(OperationOrResponseInternals));
+                var waitForCompleteMethodName = operation.ResultType != null ? "WaitForCompletionAsync" : "WaitForCompletionResponseAsync";
+
+                using (writer.Scope($"{operation.Declaration.Accessibility} partial class {cs.Name}: {baseType}"))
+                {
+                    writer.Line($"private readonly {helperType} _operation;");
+
+                    writer.Line();
+                    writer.WriteXmlDocumentationSummary($"Initializes a new instance of {cs.Name} for mocking.");
+                    using (writer.Scope($"protected {cs.Name:D}()"))
+                    {
+                    }
+
+                    writer.Line();
+                    writer.Append($"internal {cs.Name}(");
+                    // todo: programmatically get the type of operationBase from the definition of [Resource]
+                    writer.Append($"{typeof(ResourceOperationsBase)} operationsBase, ");
+                    writer.Append($"{typeof(Response)}<{operation.ResultDataType}> {responseVariable}");
+                    writer.Line($")");
+
+                    using (writer.Scope())
+                    {
+                        writer.Append($"_operation = new {helperType}(");
+                        if (operation.ResultType != null)
+                        {
+                            writer.Append($"{typeof(Response)}.FromValue(");
+                            writer.Append($"new {operation.ResultType}(operationsBase, {responseVariable}.Value),");
+                            writer.Append($"{responseVariable}.GetRawResponse()");
+                            writer.Append($")");
+                        }
+                        writer.Line($");");
+
+                    }
+
+                    writer.WriteXmlDocumentationInheritDoc();
+                    writer.Line($"public override string Id => _operation.Id;");
+                    writer.Line();
```

Comment:
These property definitions (line 69~109) are exactly the same as in `LongRunningOperationWriter`. I feel like it needs to be refactored.

Reply:
Created ADO item to track this. [https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5847](https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5847)